src/lightning/fabric/strategies/deepspeed.py

```diff
@@ -498,7 +498,7 @@ def load_checkpoint(
             )
         engine = engines[0]
 
-        from deepspeed.runtime import DeepSpeedOptimizer
+        from deepspeed.runtime.base_optimizer import DeepSpeedOptimizer
```

Comment:
Since this was a recent breaking change in deepspeed, if I see correctly in version 0.14.1, let's make the import conditioned on the version being used:

In L46, define:

```
_DEEPSPEED_GREATER_EQUAL_0_14_1 = RequirementCache("deepspeed>=0.14.1")
```

Then in the import here:
```suggestion
		if _DEEPSPEED_GREATER_EQUAL_0_14_1:
        	from deepspeed.runtime.base_optimizer import DeepSpeedOptimizer
        else:
        	from deepspeed.runtime import DeepSpeedOptimizer
```

Reply:
Great